scripts/setup_osx.sh

```diff
@@ -1,38 +1,265 @@
+}
+
+# red text
+print_error() {
+  echo -e "\033[0;31m${@}\033[0m"
+}
+
+# yellow text
+print_warning() {
+  echo -e "\033[0;33m${@}\033[0m"
+}
+
+# List of verified versions of OS X where garage has been installed
+VALID_VER=()
+VER="$(sw_vers -productVersion)"
+
+if [[ ! " ${VALID_VER[@]} " =~ " ${VER} " ]]; then
+  print_warning "It has not been verified whether garage will install" \
+    "correctly under your current OS X version (${VER}). If the installation" \
+    "is successful, please update the list of valid versions in this script." \
+    "Otherwise, if you are able to fix your compatibility issue, please file" \
+    "an issue to (https://github.com/rlworkgroup/garage/issues) and include a"\
+    "link to your Pull Request.\n" |
+  fold -s
+  while [[ "${continue_var}" != "y" ]]; do
+    read -p "Continue? (y/n): " continue_var
+    if [[ "${continue_var}" = "n" ]]; then
+      exit
+    fi
+  done
 fi
 
-echo "Creating conda environment..."
+# Verify there's a file in the mjkey path
+test -f "${_arg_mjkey_path}" || _PRINT_HELP=yes die \
+  "The path ${_arg_mjkey_path} of the MuJoCo key is not valid." 1
+
+# Make sure that we're under the garage directory
+GARAGE_DIR="$(dirname $(script_dir_path))"
+cd "${GARAGE_DIR}"
+
+# File where environment variables are stored
+BASH_PROF="${HOME}/.bash_profile"
+
+# Install dependencies
+echo "Installing garage dependencies"
+echo "You will probably be asked for your sudo password"
+
+# Homebrew is required first to install the other dependencies
+hash brew 2>/dev/null || {
+  # Install the Xcode Command Line Tools
+  xcode-select --install
+  # Install Homebrew
+  /usr/bin/ruby -e "$(curl -fsSL \
+    https://raw.githubusercontent.com/Homebrew/install/master/install)"
+}
+
+# For installing garage: bzip2, git, glfw, unzip, wget
+# For building glfw: cmake
+# Required for OpenAI gym: cmake boost boost-python ffmpeg sdl2 swig wget
+# Required for OpenAI baselines: cmake openmpi
+brew update
+brew install \
+  bzip2 \
+  git \
+  glfw \
+  unzip \
+  wget \
+  cmake \
+  boost \
+  boost-python \
+  ffmpeg \
+  sdl2 \
+  swig \
+  openmpi
+
+# Leave a note in ~/.bashrc for the added environment variables
+if [[ "${_arg_set_envvar}" = on ]]; then
+  echo -e "\n# Added by the garage installer" >> "${BASH_PROF}"
+fi
+
+# Set up MuJoCo
+if [[ ! -d "${HOME}"/.mujoco/mjpro150 ]]; then
+  mkdir "${HOME}"/.mujoco
+  MUJOCO_ZIP="$(mktemp -d)/mujoco.zip"
+  wget https://www.roboti.us/download/mjpro150_osx.zip -O "${MUJOCO_ZIP}"
+  unzip -u "${MUJOCO_ZIP}" -d "${HOME}"/.mujoco
+else
+  print_warning "MuJoCo is already installed"
+fi
+# Configure MuJoCo as a shared library
+export LD_LIBRARY_PATH="$LD_LIBRARY_PATH:${HOME}/.mujoco/mjpro150/bin"
+LD_LIB_ENV_VAR="LD_LIBRARY_PATH=\"\$LD_LIBRARY_PATH:${HOME}/.mujoco/mjpro150"
+LD_LIB_ENV_VAR="${LD_LIB_ENV_VAR}/bin\""
+if [[ "${_arg_set_envvar}" = on ]]; then
+  echo "export ${LD_LIB_ENV_VAR}" >> "${BASH_PROF}"
+fi
+
+# Set up conda
+hash conda 2>/dev/null || {
```

Comment:
don't install conda if it already exists

Reply:
If the exit status of `hash conda` is not zero, that is, conda is not installed, then the commands between the brackets are executed. Otherwise, if the exit status is zero (conda is installed), they're not executed.

Reply:
Actually not even conda works inside the subshell. I don't know if this may be related to not using PATH but the script suggested by conda.

Reply:
If I add the binaries of conda in the PATH, calling conda inside the script works again

Reply:
I took a look at this problem, and it's actually really difficult to detect a conda installation from a shell script by inspecting just the shell environment. Considering #32 I don't think we need to address this in general.

I think it's sufficient to detect `[[ -d "${HOME}/miniconda2" ]]`. If that exists, then you can `. ${HOME}/miniconda2/etc/profile.d/conda.sh` to use the conda commands. If it doesn't just install it and use it.

Expert users with more than one conda install will just have to be careful to not use `--modify-bashrc`. 

If you want to be really kind to experts, you could add the arg `--use-conda` which takes the path to an existing conda, but I think that's overkill. If someone it juggling multiple conda installs, they should be enough of an expert to modify the script.

Reply:
The problem with using PATH is that conda explicitly discourages it, to the point of displaying a nag message on every call. There is probably a reason for that. We also can't detect whether conda is installed anyway, we can only guess at the path (and it could be installed in many places).

Reply:
Got it, I will implement this approach  then.